### ISSUE TITLE
False positive no-value-for-parameter in metaclass

### ISSUE THREAD

**Issue by mrginglymus** [PyCQA/pylint#3268]
### Steps to reproduce
given `dang.py`
```python
# pylint: disable=missing-module-docstring,blacklisted-name,invalid-name,missing-function-docstring,missing-class-docstring

class MetaDang(type):

    @staticmethod
    def foo(fn, string):
        fn(string)

    def bar(cls, string):
        cls.foo(print, string)

    def baz(cls):
        cls.bar('woo')

class Dang(metaclass=MetaDang):
	pass

Dang.baz()
```

### Current behavior
```
py -3.7 -m pylint dang.py
************* Module dang
dang.py:13:8: E1120: No value for argument 'string' in unbound method call (no-value-for-parameter)

------------------------------------------------------------------
Your code has been rated at 5.00/10
```

### Expected behavior
```
py -3.7 -m pylint dang.py

-------------------------------------------------------------------
Your code has been rated at 10.00/10
```

### pylint --version output

pylint 2.4.0
astroid 2.3.3
Python 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 20:34:20) [MSC v.1916 64 bit (AMD64)]

**Comment by PCManticore** [comment]
Thanks, I agree we should fix this.

**Comment by ifinik** [comment]
Hi, it this issue, isn't it?
`from abc import ABCMeta
class Foo(ABCMeta):
    def __getattr__(cls, bar):
        return cls(bar)

class Bar(metaclass=Foo):
    def __init__(self, foo):
        self.foo = foo
`
No value for argument 'bases' in constructor call
No value for argument 'namespace' in constructor call

If "ABCMeta" is replaced by "type"
messages do not appear

**Comment by jourdanrodrigues** [comment]
Same issue here (I know I shouldn't comment, but I feel it won't get attention if they don't notice it).

**Comment by huntfx** [comment]
If the fix happens to be checking for certain inheritances (eg. `type`), I'd like to add that `sqlalchemy.ext.declarative.DeclarativeMeta` is specifically for metaclasses too.

**Comment by csuvsaregal** [comment]
I just join to this because I have the same issue on VSCode ...

I receive the error msg: No value for argument ‘url’ in unbound method call pylint(no-value-for-parameter)
Actually the error messages on bound methods using self <self.CORE.getUrl(url)> in a bound classmethod. There are more lines in the code like this without error. Some of the lines (9 in a 950 lines file) are underlined and trigger pylint. 

pylint --version output:

pylint 2.8.0
astroid 2.6.6
Python 3.9.6 (default, Jun 29 2021, 06:20:32)
[Clang 12.0.0 (clang-1200.0.32.29)]

IDE: VsCode version 1.51 (2020 October)

**Comment by DUOLabs333** [comment]
You also get this problem with `hybrid_property` from sqlalchemy.
